### ISSUE TITLE
Vector database optimized Docker images

### ISSUE THREAD

**Issue by masterkain** [openai/chatgpt-retrieval-plugin#292]
Hello everyone,

tl;dr 👉 [icoretech/chatgpt-retrieval-plugin-docker](https://github.com/icoretech/chatgpt-retrieval-plugin-docker)

I have set up a matrix build to generate Docker images for the ChatGPT Retrieval Plugin, and I would like to invite your participation in testing and providing feedback.

The Docker images have the following features:

- **Public Availability:** The Docker images are publicly accessible, allowing easy adoption by anyone interested.

- **Optimized for Vector Databases:** Each Docker image is specifically optimized for a different vector database provider. This optimization ensures optimal performance and compatibility.

- **GitOps Compatibility:** The Docker images are thoughtfully tagged, making them compatible with GitOps practices such as Flux 2. This allows for auto updates into your deployment workflows.

To the OpenAI team, if you find these Docker images useful, please feel free to mention them or include them in relevant resources.

Cheers!

**Comment by masterkain** [comment]
```
[[package]]
name = "pinecone-client"
version = "2.2.1"
description = "Pinecone client and SDK"
category = "main"
optional = false
python-versions = ">=3.6"
files = [
    {file = "pinecone-client-2.2.1.tar.gz", hash = "sha256:0878dcaee447c46c8d1b3d71c854689daa7e548e5009a171780907c7d4e74789"},
    {file = "pinecone_client-2.2.1-py3-none-any.whl", hash = "sha256:6976a22aee57a9813378607506c8c36b0317dfa36a08a5397aaaeab2eef66c1b"},
]

[package.dependencies]
dnspython = ">=2.0.0"
loguru = ">=0.5.0"
numpy = "*"
python-dateutil = ">=2.5.3"
pyyaml = ">=5.4"
requests = ">=2.19.0"
tqdm = ">=4.64.1"
typing-extensions = ">=3.7.4"
urllib3 = ">=1.21.1"
```

currently `loguru` seems to be a direct dep of `pinecone-client`, so if we remove the package we get:

```
INFO:     Started server process [7]
INFO:     Waiting for application startup.
ERROR:    Traceback (most recent call last):
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 671, in lifespan
    async with self.lifespan_context(app):
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 566, in __aenter__
    await self._router.startup()
  File "/usr/local/lib/python3.10/site-packages/starlette/routing.py", line 648, in startup
    await handler()
  File "/code/./server/main.py", line 151, in startup
    datastore = await get_datastore()
  File "/code/./datastore/factory.py", line 24, in get_datastore
    from datastore.providers.weaviate_datastore import WeaviateDataStore
  File "/code/./datastore/providers/weaviate_datastore.py", line 8, in <module>
    from loguru import logger
ModuleNotFoundError: No module named 'loguru'
```
